### DISCUSSION TITLE
Azure with supabase instead of s3

### DISCUSSION THREAD

**Discussion by GaushJMeta** [supabase/supabase#20045] [Questions]
Hello, I have a client who's using azure blob storage. Can we integrate azure storage with supabase? If yes then how? Thanks!

**Comment by Peterksharma** [comment]
Is there any updates on this?

**Comment by jmealo** [comment]
Since only a single API is supported right now (S3) it's unclear as a contributor where ABS support should live and how it'd be configured/tested? 🤔 

I'm assuming we'd need CI/CD infra emulating ABS:
https://github.com/marketplace/actions/azuright

So we would ideally have all existing tests run against both S3 and ABS to make sure there's no regressions.

I'm not sure if it makes sense to use the Azure Node.js client (if it's autogenerated garbage) or if it's a decent client. I suppose that needs to be decided on.
